src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -925,7 +928,7 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
         case other =>
           fatal(
             s"""method 'annotate_variants_table' expects a key table keyed by one of the following:
-               |  [ Variant ]
+               |  [ $vSignature ]
```

Comment:
we basically need to build rules about which types can be promoted to other types